src/nccl_ofi_rdma.cc

```diff
@@ -3,32 +3,150 @@
  */
 #include "config.h"
 
+#include <algorithm>
 #include <assert.h>
 #include <inttypes.h>
+#include <memory>
+#include <optional>
+#include <string>
 #include <stdio.h>
 #include <sys/mman.h>
 #include <unistd.h>
 #include <pthread.h>
 #include <stdlib.h>
+#include <array>
+#include <mutex>
+#include <unordered_map>
+#include <unordered_set>
+#include <string_view>
 
 #include "nccl_ofi.h"
 #if HAVE_CUDA
 #include "nccl_ofi_cuda.h"
 #endif
-#include "nccl_ofi_ep_addr_list.h"
 #include "nccl_ofi_param.h"
 #include "nccl_ofi_rdma.h"
 #include "nccl_ofi_math.h"
 #include "nccl_ofi_tracepoint.h"
 #include "nccl_ofi_scheduler.h"
 #include "nccl_ofi_topo.h"
 #include "nccl_ofi_kvstore.h"
-#include "nccl_ofi_memcheck.h"
 #include "nccl_ofi_ofiutils.h"
 #include "nccl_ofi_pthread.h"
 #include "nccl_ofi_dmabuf.h"
 #include "nccl_ofi_mr.h"
 
+/*
+ * @brief	Endpoint rail
+ *
+ * Endpoint rail encapsulates data of an endpoint for a
+ * specific rail.
+ */
+struct nccl_net_ofi_ep_rail {
```

Comment:
I don't think we should make this change.  We want to split up rdma.c, not make it harder to split up later.

Reply:
Even in light of the conversation below, I still have strong feelings on this one.  This is not a useful change for us.

Reply:
Can we put the ep map code into a new file nccl_ofi_rdma_ep_map.{h,c} or some such?